packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -377,12 +384,34 @@ class CreateCommand extends FlutterCommand with CreateBase {
       includeLinux = false;
       includeMacos = false;
       includeWindows = false;
+      includeDarwin = false;
     } else {
-      includeIos = featureFlags.isIOSEnabled && platforms.contains('ios');
+      final bool darwinRequested = platforms.contains('darwin');
+      final bool darwinSupported =
+          (template == FlutterTemplateType.plugin) &&
+          featureFlags.isIOSEnabled &&
+          featureFlags.isMacOSEnabled;
+
+      if (darwinRequested && darwinSupported) {
+        includeDarwin = true;
+        includeIos = featureFlags.isIOSEnabled;
+        includeMacos = featureFlags.isMacOSEnabled;
+      } else if (darwinRequested && !darwinSupported) {
+        includeDarwin = false;
+        globals.printWarning(
+          'Warning: To use the "darwin" platform, you must have both iOS and macOS enabled.\n'
+          'Run "flutter config --enable-ios --enable-macos-desktop" and try again.',
+        );
```

Comment:
Current `darwin` is included in `platforms` by default I think. So when you do `flutter create my_app`, you'll get this warning, which doesn't make sense when creating an app